content/docs/components/icon-button/theming.mdx

```diff
@@ -2,3 +2,198 @@
 id: icon-button
 scope: theming
 ---
+
+The `IconButton` component is a single part component. All of the styling is applied
+directly to the `Button` component.
+
+
+> To learn more about styling single part components, visit the
+> [Component Style](/docs/styled-system/component-style#styling-single-part-components)
+> page.
+
+## Theming properties
+
+The properties that affect the theming of the `IconButton` component are:
+
+- `variant`: The visual variant of the button. Defaults to `solid`.
+- `colorScheme`: The color scheme of the button. Defaults to `gray`.
+- `size`: The size of the button. Defaults to `md`.
+
+## Theming utilities
+
+- `defineStyle`: a function used to create style objects.
+- `defineStyleConfig`: a function used to define the style configuration for a
+  single part component.
+
+
+```jsx live=false
+import { defineStyle, defineStyleConfig } from '@chakra-ui/react'
+```
+
+## Customizing the default theme
+
+```jsx live=false
+import { defineStyle, defineStyleConfig } from '@chakra-ui/react'
+
+const outline = defineStyle({
+  border: '2px dashed', // change the appearance of the border
+  borderRadius: 0, // remove the border radius
+  fontWeight: 'semibold', // change the font weight
+})
+
+
+const xl = defineStyle({
+  fontSize: 'xl',
+  px: '6',
+  h: '16',
+  borderRadius: 'md',
+})
+
+export const buttonTheme = defineStyleConfig({
+  variants: { outline },
+})
+```
+
+After customizing the button theme, we can import it in our theme file and add
+it in the `components` property:
+
+```jsx live=false
+import { extendTheme } from '@chakra-ui/react'
+import { buttonTheme } from './components/button'
+
+export const theme = extendTheme({
+  components: { Button: buttonTheme },
+})
+```
+
+> The `IconButton` and `Button` components share the same theming configuration.
+> To create theming just for the `IconButton` component, you can create a **custom variant**.
```

Comment:
Let's push this at the top of the doc. I think it's better to give the users the heads up as early as possible.